### PR TITLE
Update docs related to supported python version [ENG 199]

### PR DESCRIPTION
**Pull request by varunshankar**
**Proposed changes**:
- As part of this task, it is decided not to do any dependency upgrades but update the docs.
- Fixes: https://rasahq.atlassian.net/browse/ENG-199
**Status (please check what you already did)**:
- [ ] added some tests for the functionality
- [x] updated the documentation
- [ ] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/main/changelog) for instructions)
- [ ] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

File: docs/docs/installation/environment-set-up.mdx [hunk 1]
```diff
@@ -16,9 +16,6 @@ pip3 --version
 :::note Python3 Supported Versions
 
 Currently, rasa supports the following Python versions: `3.7`, `3.8`, `3.9` and `3.10`.
-Note that Python `3.10` is not currently supported for rasa installations on Apple Silicon.
-The blocker consists of current inability to update to Tensorflow version >= `2.10.x` which is pending the resolution
-of a Tensorflow bug.
```
Comment (ancalita) [inline docs/docs/installation/environment-set-up.mdx:21]: We need to specify here that  Python `3.10` on Apple Silicon doesn't work in `3.4.x`, but will be supported from `3.5.x` onwards.

File: docs/docs/installation/environment-set-up.mdx [hunk 2]
```diff
@@ -16,9 +16,6 @@ pip3 --version
 :::note Python3 Supported Versions
 
 Currently, rasa supports the following Python versions: `3.7`, `3.8`, `3.9` and `3.10`.
```
Comment (ancalita) [inline docs/docs/installation/environment-set-up.mdx:18]: We also need to mention that `3.10` is only available from `3.4.x` onwards.

File: docs/docs/installation/environment-set-up.mdx [hunk 3]
```diff
@@ -16,9 +16,8 @@ pip3 --version
...
-Note that Python `3.10` is not currently supported for rasa installations on Apple Silicon.
-The blocker consists of current inability to update to Tensorflow version >= `2.10.x` which is pending the resolution
-of a Tensorflow bug.
+It's worth noting that Python `3.10` is only available for versions `3.4.x` onwards.
```
Comment (ancalita) [inline docs/docs/installation/environment-set-up.mdx:19]:
```suggestion
Note that Python `3.10` is only supported for versions `3.4.x` and upwards.
```
Hopefully `upwards` is not considered a typo also 😅

File: docs/docs/installation/environment-set-up.mdx [hunk 4]
```diff
@@ -16,9 +16,8 @@ pip3 --version
...
+Although, on Apple Silicon, Python `3.10` isn't functional in `3.4.x` but will be supported starting from `3.5.x`.
```
Comment (ancalita) [inline docs/docs/installation/environment-set-up.mdx:20]:
```suggestion
Additionally, rasa installation on Apple Silicon with Python `3.10` is not functional in `3.4.x` but will be supported starting from `3.5.x`.
```

**Comment by varunshankar** [comment]
Thank you for the suggestions @ancalita :)

**Review by ancalita — APPROVED** [review]

**Pull request merged**